tracing/jaeger/src/main/java/io/helidon/tracing/jaeger/JaegerTracerBuilder.java

```diff
@@ -513,4 +520,13 @@ String config() {
             return config;
         }
     }
+
+    private static boolean checkAgent() {
+        try {
+            Class.forName("io.opentelemetry.javaagent.bootstrap.AgentStarter");
```

Comment:
As we discussed yesterday, this approach handles the single situation in which the OpenTelemetry agent is present (and has established the global tracer itself) but there could be various other cases in which other code (perhaps user code), for whatever reason, has assigned the global tracer before our code has a chance to do so. The OTel agent being present is just one such case.

Is there a way to create a more general solution? 

From my quick look I did not see a way to find out simply if the global `OpenTelemetry` has been set. We can `get` it (OTel automatically creates a no-op instance if `set` _had not_ previously been invoked, which does not work focus) or `set` it (OTel throws an exception if `set` _had_  previously been invoked).

But could we use that behavior to our advantage? I have not tried this, but instead of checking for the presence of the agent, what if the Helidon code which invokes `GlobalOpenTelemetry#set` caught  the `IllegalStateException`,  logged a message indicating that the global OTel has already been set, and invokes `get` to retrieve that previously-set instance?

That would tell the user that any configuration she had used which _Helidon_ works would not have been used in setting up OT and would allow our code to continue, using either the OTel instance we had built and set _or_ the previously-set one. 

The exception which OT throws contains the stack trace captured when `GlobalOpenTelemetry#set` was invoked and we could include that in our log message (perhaps at `FINE` level) so the user understand what code had set the global OTel instance.

This might be a little more code than what's in the PR currently, but 

- it would be less fragile (OTel could change the agent class name whereas it seems less likely they would change the signature and behavior of `GlobalOpenTelemetry#set`, and
- it would handle any case, not just the agent case, in which other code has set the global OT instance before our code does.

Reply:
Please never ever use `Class.forName("")` in Helidon.
We should handle this type of requirements differently

Reply:
Currenty using fast solution with indirect checks of properties and instrumented `Context`

Reply:
Fixed